### PR TITLE
Update docs about composer-packaged themes to include naming convention

### PR DESCRIPTION
**Pull request by DrewML**
## Purpose of this pull request

Make some implicit theme requirements explicit in the docs.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/extension-dev-guide/package/package_module.html
- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/themes/theme-structure.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/frontend-dev-guide/themes/theme-structure.md [hunk 1]
```diff
@@ -230,7 +230,7 @@ The directories and files structure described below is the most extended one. It
       </td>
       <td colspan="1">optional</td>
       <td colspan="1">
-        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package.
+        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>
```
Comment (jeff-matthews) [inline guides/v2.1/frontend-dev-guide/themes/theme-structure.md]:
```suggestion
        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>.
```

File: guides/v2.2/frontend-dev-guide/themes/theme-structure.md [hunk 2]
```diff
@@ -230,7 +230,7 @@ The directories and files structure described below is the most extended one. It
       </td>
       <td colspan="1">optional</td>
       <td colspan="1">
-        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package.
+        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>
```
Comment (jeff-matthews) [inline guides/v2.2/frontend-dev-guide/themes/theme-structure.md]:
```suggestion
        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>.
```

File: guides/v2.3/frontend-dev-guide/themes/theme-structure.md [hunk 3]
```diff
@@ -230,7 +230,7 @@ The directories and files structure described below is the most extended one. It
       </td>
       <td colspan="1">optional</td>
       <td colspan="1">
-        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package.
+        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>
```
Comment (jeff-matthews) [inline guides/v2.3/frontend-dev-guide/themes/theme-structure.md]:
```suggestion
        Describes the theme dependencies and some meta-information. Will be here if your theme is a Composer package. The "name" field must be in the format <code>"&lt;vendor-name&gt;/theme-&lt;area&gt;-&lt;theme-name&gt;"</code>.
```

**Comment by DrewML** [comment]
Done - thanks for making it easy!

**Comment by jeff-matthews** [comment]
Thanks for updating the docs!

**Pull request merged**